src/aggregations/aggregation-builder.js

```diff
@@ -0,0 +1,64 @@
+import _ from 'lodash'
+import aggregations from './index'
+
+const AggregationBuilder = {
+  /**
+   * Apply a aggregation of a given type providing all the necessary arguments,
+   * passing these arguments directly to the specified aggregation builder.
+   * Merges existing aggregation(s) with the new aggregation.
+   *
+   * @param  {String}  type Name of the aggregation type.
+   * @param  {...args} args Arguments passed to aggregation builder. May include
+   *                        a nest function as the last item.
+   * @returns {AggregationBuilder} Builder object.
+   */
+  aggregation(type, ...args) {
+    let klass = aggregations[type]
+    let aggregation = Object.create(AggregationBuilder)
+
+    if (!klass) {
+      throw new TypeError(`Aggregation type ${type} not found.`)
+    }
+
+    // If last argument is a nesting function, remove it
+    // from `args` before building the aggregation
+    let nest = _.last(args)
+    args = _.isFunction(nest) ? _.initial(args) : args
+
+    // Mixin to assign newly built aggregation properties
+    // to the `AggregationBuilder`
+    _.extend(aggregation, klass(...args))
+
+    // Extend the current aggregation object with the
+    // recently built aggregation
+    this._aggs = _.assign(this._aggs, aggregation)
+
+    if (_.isFunction(nest)) {
+      // Resolve the nested aggregation and set it as a
+      // children of the current aggregation
+      this._aggs[_.findKey(aggregation)].aggs = nest(aggregation)._aggs
+    }
+    return this
+  },
+
+  /**
+   * Alias for `aggregation`.
+   *
+   * @param {...[Object]} args Arguments for `aggregation`.
+   * @returns {AggregationBuilder} Builder object.
+   */
+  add(...args) {
```

Comment:
It's not, really. But it makes for a nicer syntax when using the builder inside the main `BodyBuilder` class: https://github.com/danpaz/bodybuilder/pull/16/files#diff-1fdf421c05c1140f6d71444ea2b27638R270

Shall I remove it?